scripts/deploy.js

```diff
@@ -0,0 +1,160 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { spawn } = require('child_process');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const WORKSPACE_PRODUCTION = 'Production';
+const WORKSPACE_STAGE = 'Stage';
+const WORKSPACE_QA = 'QA';
+const WORKSPACE_DEVELOPMENT = 'Development';
+
+function execCommand(command, args) {
+  return new Promise((resolve, reject) => {
+    const cmd = spawn(command, args);
+
+    let output = '';
+    let error = '';
+
+    cmd.stdout.on('data', (data) => {
+      output += data.toString();
+    });
+    cmd.stderr.on('data', (data) => {
+      error += data.toString();
+    });
+
+    cmd.on('close', (code) => {
+      if (code !== 0) {
+        reject(new Error(`Command exited with code ${code}: ${error}`));
+      } else {
+        resolve(output);
+      }
+    });
+  });
+}
+
+function convertToWorkspaceName(branchName) {
+  return branchName.replace(/[^a-zA-Z0-9]/g, '');
+}
+
+async function getCurrentGitBranch() {
+  try {
+    const { stdout } = await exec('git rev-parse --abbrev-ref HEAD');
+    return stdout.trim().toLowerCase();
+  } catch (error) {
+    throw new Error("Error fetching Git branch. Ensure you're in a Git repository.");
+  }
+}
+
+async function selectWorkspace(workspaceName) {
+  console.log(`Selecting workspace: ${workspaceName}...`);
+  try {
+    await execCommand('aio', ['app', 'use', '-w', workspaceName, '--no-service-sync', '--merge']);
+    console.log(`Workspace '${workspaceName}' selected successfully.`);
+  } catch (error) {
+    throw new Error(`Failed to select workspace: ${error.message}`);
+  }
+}
+
+async function createWorkspace(workspaceName) {
+  console.log(`Creating new workspace: ${workspaceName}...`);
+  try {
+    await execCommand('aio', ['app', 'use', '-w', workspaceName, '--no-confirm-new-workspace', '--no-service-sync', '--merge']);
+    console.log(`Workspace '${workspaceName}' created successfully.`);
+  } catch (error) {
+    throw new Error(`Failed to create workspace: ${error.message}`);
+  }
+}
+
+function getDeploymentUrl(workspaceName) {
+  if (WORKSPACE_PRODUCTION === workspaceName) {
+    return 'https://experience.adobe.com/#/aem/generate-variations/';
+  } else if (WORKSPACE_STAGE === workspaceName) {
+    return 'https://experience-stage.adobe.com/?shell_ims=prod#/aem/generate-variations/';
+  } else if (WORKSPACE_QA === workspaceName) {
+    return 'https://experience-qa.adobe.com/?shell_ims=prod#/aem/generate-variations/';
+  } else if (WORKSPACE_DEVELOPMENT === workspaceName) {
+    return 'https://experience-qa.adobe.com/?devMode=true&shell_source=dev&shell_ims=prod#/aem/generate-variations/';
+  }
+  return `https://experience-qa.adobe.com/?devMode=true&shell_source=dev&shell_ims=prod&app_workspace=${workspaceName}#/aem/generate-variations/`;
```

Comment:
`devMode=true` and `shell_source=dev` are unnecessary here for the QA URL.

Reply:
I would keep it there as it is supposed to help with CSP: https://git.corp.adobe.com/pages/exc/unified-shell-docs/docs/development/query-params#devmode